### PR TITLE
￼ gstreamer1: update to 1.18.2

### PR DESCRIPTION
**Pull request by ndowens**
gst1-python version 1.18.1 doesn't support python2 anymore so can not update it

If someone can test update please let me know if update works fine.

For me, no matter gst version I have installed I can't not get audio out from an app that depends on gstreamer

### CODE REVIEW

File: srcpkgs/gstreamer1/template
Comment (fosslinux) [inline srcpkgs/gstreamer1/template:13]: Why was this rmeoved?

**Review by fosslinux — CHANGES_REQUESTED** [review]

File: srcpkgs/gst-libav/template
Comment (ericonr) [inline srcpkgs/gst-libav/template:25]: is the gcc bug fixed? or did they change their compile flags?

File: srcpkgs/gst-plugins-bad1/template
Comment (ericonr) [inline srcpkgs/gst-plugins-bad1/template:7]: Why change? Is this a new xlint?

File: srcpkgs/gst-omx/patches/0001-ARM-build.patch
Comment (ericonr) [inline srcpkgs/gst-omx/patches/0001-ARM-build.patch:4]: Can you explain more clearly what this patch is doing?

File: srcpkgs/gstreamer1/template
Comment (ndowens) [reply srcpkgs/gstreamer1/template:13]: Doesn't actually build even if enabled, and option was correct, so it is something with upstream

File: srcpkgs/gst-omx/patches/0001-ARM-build.patch
Comment (q66) [reply srcpkgs/gst-omx/patches/0001-ARM-build.patch:4]: seems to be a bunch of ugly hacks for the rpi target, i'd just drop it and use `-Dtarget=generic` universally (i.e. drop the whole `XBPS_TARGET_MACHINE` switch case)

File: srcpkgs/gstreamer1/template
Comment (ericonr) [reply srcpkgs/gstreamer1/template:13]: Why remove it, then? Just remove it as default for now, ask upstream to fix. And definitely leave a comment in the template and in the commit message about it.

Comment (q66) [reply srcpkgs/gstreamer1/template:13]: it's probably fine to remove, especially considering gtkdoc is fucky with crosstargets; i doubt that anybody will read local gtkdoc html documentation either way

File: srcpkgs/gst-libav/template
Comment (ndowens) [reply srcpkgs/gst-libav/template:25]: Added it in meson.build just in case it does still exist, as the file that is being sed'd doesn't exist now.

Comment (q66) [reply srcpkgs/gst-libav/template:25]: the `-O2` was for `x86_64-musl` only, you should test if it builds without the patch on that target and if yes, then there's no need for a patch

Comment (ndowens) [reply srcpkgs/gst-libav/template:25]: It built fine w/o -O2 on x86_64-musl, so will remove patch then

Comment (ericonr) [reply srcpkgs/gst-libav/template:25]: It would be nice to test it at runtime as well; the bug might not have been compile time.

Comment (q66) [reply srcpkgs/gst-libav/template:25]: that's highly unlikely, this wording implies some kind of ICE or similar condition

**Comment by ericonr** [comment]
Please try removing `python` from `hostmakedepends` and use `python3` instead.

Your commit message still doesn't say that you removed the build option from one of the templates.

**Comment by lane-brain** [comment]
Would love to see this happen

**Comment by ndowens** [comment]
> Please try removing `python` from `hostmakedepends` and use `python3` instead.
> 
> Your commit message still doesn't say that you removed the build option from one of the templates.

The one I see I removed option from, gstreamer1, does have "Remove gtk_doc option as it does nothing and doubt many use" Unless I am blind, I dont see 'python' but python3

**Comment by fosslinux** [comment]
> Please try removing python from hostmakedepends and use python3 instead.

This has already bene completed.

**Comment by the-eater** [comment]
the gstreamer devs are angry with Void because of the `sndio` plugin and them getting bug reports about it, would it be possible to be more clear that the sndio plugin is not maintained by them?

**Comment by ndowens** [comment]
Not sure the best way would to go about that. Wait for a member of Void to respond

**Comment by fosslinux** [comment]
What am I missing? The plugin seems to come through their official tarball, how is it not maintained by them?

@the-eater do you have a ml link or something

**Comment by ericonr** [comment]
@fosslinux we have a 1k line patch for adding sndio support.

**Comment by the-eater** [comment]
> What am I missing? The plugin seems to come through their official tarball, how is it not maintained by them?
> 
> @the-eater do you have a ml link or something

https://twitter.com/sdroege_/status/1340342884197920770

here's one of the devs talking about it

**Comment by q66** [comment]
this is quite bad, the patch should be either removed entirely, or separated into a "you're on your own" package... the official stance of void has always been not to patch in features, and it's even worse if it causes actual bugs for other people

so before we merge this, it needs to be dealt with

**Comment by ndowens** [comment]
Removed sndio patch from gst-plugins-base1

**Comment by ericonr** [comment]
I'd prefer the solution of splitting it or making it a build option... We don't usually patch things, but sndio support has been consistently added throughout our packages.

I can try to take a look at the patch, but that will happen later; we should be careful with OpenBSD patches for sndio because they don't always implement proper fallback.

**Comment by q66** [comment]
what has been consistently asked for does not matter - we're not the upstream, so we have no responsibility for maintaining entire components added on top, and as you can see, if we do that it leads to unfortunate incidents with upstream projects; i *really* don't want to become another gentoo, where our users come to upstreams with bug reports caused by their outlandish setups

if people want those features available, they should go to upstream, open reports and PRs there, and *then* we can pull it downstream, but maintaining entire new features within a distro is really bad

**Comment by fosslinux** [comment]
I strongly agree with @q66. Why does sndio become the exception for pulling in features? Unlike libressl (for example), we have alternatives (such as pulseaudio), and patching this in with a 1k LOC patch and a ton of new files is not very good.

This patch was originally added in #8056 and has barely changed since then. upstream is not hostile to readding this patch, it's just that there was no maintainer (src: https://twitter.com/sdroege_/status/1340367227581829123). If someone is interested in putting that work in, then upstream will merge it and we will take it *as downstream*, as our policy is in general.

I honestly think while sndio is a great project there is not enough upstream support and patches are not being maintained. Other quite large patches include:

- portaudio - https://github.com/void-linux/void-packages/blob/master/srcpkgs/portaudio/patches/sndio.patch (883 lines)
- chromium - https://github.com/void-linux/void-packages/commit/293c8a41e6ab5c349db794ad748b00ba66c1a21c (886 lines)
- SDL - https://github.com/void-linux/void-packages/blob/master/srcpkgs/SDL/patches/SDL-1.2.15-sndio.patch (375 lines)

There are many other smaller patches which IMO are fine as they are just altering build systems, including:

- baresip
- firefox{,-esr}
- icecat
- qt5-webengine
- thunderbird

Futhermore sndio seems to be "prioritized" in a lot of packages over alsa. (?!?!?!?!?)

**Comment by q66** [comment]
there should also be an alsa plugin for sndio, so people should just be able to use the alsa backend in apps and sndio should be able to pick them up

**Comment by ericonr** [comment]
The alsa plugin has issues with syncing (watching videos with it is quite terrible, from what I've heard), and it doesn't work for audio inputs.

The portaudio patch was fixed up a while ago to actually implement proper fallback for other sound devices. I don't know about SDL or chromium.

**Comment by travankor** [comment]
>It would suck a bit to change that policy by removing sndio support from a bunch of packages at the same time. If the current packages aren't known to cause any issues, I think it's ok to leave thing as is, as long as we have people trying to work to push these upstream.

Is this an official policy or just something that happened naturally over time? I understand that there are a lot of packages patches around to support sndio, but sndio is not really advertised or mentioned whenever Void Linux comes.

**Comment by q66** [comment]
there was no concrete decision to do this, things were largely added by random people over time, and nothing was ever advertised

**Comment by q66** [comment]
also, having large patches in ~3 projects and small ones in a couple others can hardly be called a conscious effort to make something a first class support

**Comment by ericonr** [comment]
Ok, it wasn't a policy, bad choice of words. More of a natural thing that came about.

All I'm asking is we don't have a sndio removal spree. Let it happen if complaints come about for certain packages, but otherwise leave as is.

**Comment by ericonr** [comment]
we do enable sndio on almost all projects that support it, and those patches are in widely used packages.

**Comment by q66** [comment]
again, removal/disablement of support in 2 projects (gstreamer is one, and qt5-webengine is the other) can hardly be called a "spree" and is following *actual* policy that is in place (which is, we do *not* patch in features into upstream projects)

**Comment by q66** [comment]
it's *especially* because those large patches are in widely used packages that we should not be patching them

**Comment by ericonr** [comment]
I don't want this to be a hill for me to die on; I can definitely build packages myself with those removals reverted. I should note that it *will* be a breaking change for those users who rely on said patches, though not a major one.

Portaudio having sndio also enables any application that uses it to talk to sndio, so I'd say it's a important package for the purpose of sndio support.

Anyway, this is the last I will touch the subject.

**Comment by q66** [comment]
again, if users rely on patches, it should be in their best interest to put in effort to work them into the respective upstreams, but maintaining that in distros is just wrong

i can tell you're a sndio user and therefore biased on the subject, but that is not a reason to keep potentially sketchy patches (and as a result give a poor impression to upstreams) either

**Comment by ndowens** [comment]
Currently waiting to hear from upstream why tests fail in gst-plugins-bad1; so this is the current hold-up.

**Comment by ericonr** [comment]
Tests in bad are allowed to fail; it's the ugly duck.

Just do `ninha check || msg_warn "Tests failed, not entirely unexpected"` or similar.

File: srcpkgs/gst-plugins-base1/template
Comment (ericonr) [inline srcpkgs/gst-plugins-base1/template:31]: why do they need to be outside chroot? maybe they need dbus or something?

Comment (ndowens) [reply srcpkgs/gst-plugins-base1/template:31]: Starting to believe it may need itself more so than anything. Researching errors like it was giving test, people were saying you need gst-plugins-base.

File: srcpkgs/gst-plugins-base1/template
Comment (travankor) [inline srcpkgs/gst-plugins-base1/template:31]: why did you partially undo https://github.com/void-linux/void-packages/commit/2e9ed7988c310e872cf22cba88c2886c06a66fd9?

Comment (ndowens) [reply srcpkgs/gst-plugins-base1/template:31]: Just added additional warning if someone wants to enable sndio since it isn’t to be supported upstream

**Comment by travankor** [comment]
ping?

**Comment by travankor** [comment]
@ndowens need  to rebase

I can adopt this PR if you're busy, too.

**Comment by ndowens** [comment]
> @ndowens need to rebase
> 
> I can adopt this PR if you're busy, too.

Feel free

**Comment by travankor** [comment]
#27996

**Pull request closed without merging**
